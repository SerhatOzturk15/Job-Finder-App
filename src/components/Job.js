import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import {
  Card,
  CardActionArea,
  CardActions,
  CardContent,
  CardMedia,
  Button,
  Typography,
  Container,
  CircularProgress,
} from "@material-ui/core";

const useStyles = makeStyles({
  root: {
    maxWidth: 500,
  },
  media: {
    height: 350,
    backgroundSize: "contain",
  },
});

const Job = ({
  isLoading,
  job = {},
  jobImage,
  noJobImage,
  goBack,
  goBackText,
}) => {
  const classes = useStyles();

  return (
    <>
      {isLoading ? (
        <CircularProgress
          style={{ position: "absolute", top: "30%", left: "50%" }}
          size={100}
          disableShrink
        />
      ) : (
        <Container data-testid="job-container" maxWidth="sm">
          <Card className={classes.root}>
            <CardActionArea>
              {Object.keys(job).length > 0 ? (
                <CardMedia
                  className={classes.media}
                  image={jobImage}
                  data-testid="job-found-image"
                  title="Contemplative Reptile"
                />
              ) : (
                <CardMedia
                  className={classes.media}
                  image={noJobImage}
                  data-testid="job-not-found-image"
                  title="Contemplative Reptile"
                />
              )}

              <CardContent>
                <Typography gutterBottom variant="h5" component="h2">
                  {job.title}
                </Typography>
                <Typography variant="body2" color="textSecondary" component="p">
                  {job.description}
                </Typography>
                <Typography variant="body2" color="textSecondary" component="p">
                  {job.employment_type}
                </Typography>
              </CardContent>
            </CardActionArea>
            <CardActions>
              <Button
                size="small"
                color="primary"
                onClick={() => goBack()}
                data-testid="goBackButton"
              >
                {goBackText}
              </Button>
            </CardActions>
          </Card>
        </Container>
      )}
    </>
  );
};

export default Job;
