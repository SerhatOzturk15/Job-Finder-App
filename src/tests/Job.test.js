import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Job from "../components/Job";
import { mount } from 'enzyme';

jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: jest.fn()
  })
}));

const initialState = [
    {
      "id": 1,
      "title": "Frontend Developer",
      "description": "Comfortable with modern JS stack, experience with React.",
      "employment_type": "full_time"
    },
    {
      "id": 2,
      "title": "Backend Developer",
      "description": "Comfortable with Phyton, good knowledge of databases",
      "employment_type": "full_time"
    },
    {
      "id": 3,
      "title": "Designer",
      "description": "Comfortable with photoshop. Having creative thinking ability",
      "employment_type": "part_time"
    }
  ];

const setup = () => {
  const match = { params: { id: 1 } }
  const component = shallow(<Job match={match} />);
  console.log(component.debug());
  return component;
};

describe("Job component", () => {
  let component = setup();
  it("component should render", () => {
    const wrapper = component.find('[data-testid="job-container"]');
    expect(wrapper.length).toBe(1);
  });

  it("Job snapshot should match", () => {
    const wrapper = component.find('[data-testid="job-container"]');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it ('Job Found Image should render', () => {
    const match = { params: { id: 4 } }
    const component = shallow(<Job match={match} />);
    const wrapper = component.find('[data-testid="job-not-found-image"]');
    expect(wrapper.length).toBe(1);
  });
});

