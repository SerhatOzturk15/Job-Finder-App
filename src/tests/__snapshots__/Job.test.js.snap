// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job component Job snapshot should match 1`] = `
<WithStyles(ForwardRef(Container))
  data-testid="job-container"
  maxWidth="sm"
>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-root-1"
  >
    <WithStyles(ForwardRef(CardActionArea))>
      <WithStyles(ForwardRef(CardMedia))
        className="makeStyles-media-2"
        data-testid="job-not-found-image"
        image="nojob.jpg"
        title="Contemplative Reptile"
      />
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h2"
          gutterBottom={true}
          variant="h5"
        />
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
          component="p"
          variant="body2"
        />
      </WithStyles(ForwardRef(CardContent))>
    </WithStyles(ForwardRef(CardActionArea))>
    <WithStyles(ForwardRef(CardActions))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[Function]}
        size="small"
      >
        Go Back
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(CardActions))>
  </WithStyles(ForwardRef(Card))>
</WithStyles(ForwardRef(Container))>
`;
